lang/src/org/partiql/lang/types/StaticType.kt

```diff
@@ -11,10 +11,24 @@ import org.partiql.lang.ast.passes.inference.isUnknown
 import org.partiql.lang.eval.ExprValue
 import org.partiql.lang.eval.ExprValueType
 import org.partiql.lang.eval.name
-import org.partiql.lang.eval.numberValue
 import org.partiql.lang.eval.stringValue
 import org.partiql.lang.eval.timeValue
-import java.math.BigDecimal
+import org.partiql.lang.ots.interfaces.ScalarType
+import org.partiql.lang.ots.interfaces.TypeParameters
+import org.partiql.lang.ots.plugins.standard.types.BlobType
+import org.partiql.lang.ots.plugins.standard.types.BoolType
+import org.partiql.lang.ots.plugins.standard.types.ClobType
+import org.partiql.lang.ots.plugins.standard.types.DateType
+import org.partiql.lang.ots.plugins.standard.types.DecimalType
+import org.partiql.lang.ots.plugins.standard.types.FloatType
+import org.partiql.lang.ots.plugins.standard.types.Int2Type
+import org.partiql.lang.ots.plugins.standard.types.Int4Type
+import org.partiql.lang.ots.plugins.standard.types.Int8Type
+import org.partiql.lang.ots.plugins.standard.types.IntType
+import org.partiql.lang.ots.plugins.standard.types.StringType
+import org.partiql.lang.ots.plugins.standard.types.SymbolType
+import org.partiql.lang.ots.plugins.standard.types.TimeStampType
+import org.partiql.lang.ots.plugins.standard.types.TimeType
```

Comment:
To confirm, are we on the same page that we are moving towards a design where the above references to the specific scalar types (BlobType, ..., TimeType) do not occur in this file?

Reply:
Yes. the work of designing API between scalar types interfaces & compile pipeline in `org.partiql.lang` will come later

Reply:
I'd stress again that it's important to start doing this sooner rather than later. I believe doing that would give more insight into what the plugin APIs should be than thinking ahead what are the obstacles to doing that or looking at Trino for inspiration.